client/ayon_maya/plugins/workfile_build/set_context_placeholder.py

```diff
@@ -0,0 +1,104 @@
+# -*- coding: utf-8 -*-
+
+from ayon_maya.api.workfile_template_builder import (
+    MayaPlaceholderPlugin
+)
+from ayon_maya.api.lib import set_context_settings
+from ayon_core.lib import BoolDef
+from ayon_core.lib.events import weakref_partial
+
+
+class SetContextMayaPlaceholderPlugin(MayaPlaceholderPlugin):
+    """Set context variables at different stages of workfile build.
+    This placeholder allows to run scripts at different stages of the
+    workfile build process to set context variables dynamically.
+
+    """
+
+    identifier = "maya.set_context"
+    label = "Set Context Settings"
+
+    use_selection_as_parent = False
+
+    def get_placeholder_options(self, options=None):
+        options = options or {}
+        return [
+            BoolDef(
+                "set_context",
+                label="Set Context",
+                tooltip="Enable setting context variables",
+                default=options.get("set_context", False)
+            ),
```

Comment:
I'd remove this. If they don't want to set the context, then they just don't create the placeholder.